symphony/bdk/core/retry/__init__.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import Callable
+from functools import wraps
+
+from tenacity.retry import retry_if_exception
+from tenacity import stop_after_attempt, wait_exponential, before_sleep_log
+
+from symphony.bdk.core.config.model.bdk_retry_config import BdkRetryConfig
+from symphony.bdk.core.retry.strategy import refresh_session_if_unauthorized
+
+from ._asyncio import AsyncRetrying
+
+
+def retry(*dargs, **dkw):  # noqa
+    """A decorator that provides a mechanism to to retry failed requests
```

Comment:
Please document the decorated method should be a class instance method with a valid field `self._retry_config`.